yjit/src/core.rs

```diff
@@ -3193,11 +3210,45 @@ impl<T> RefUnchecked for Cell<T> {
 mod tests {
     use crate::core::*;
 
+    #[test]
+    fn type_size() {
+        // Check that we can store types in 4 bits,
+        // and all local types in 32 bits
+        assert_eq!(mem::size_of::<Type>(), 1);
+        assert!(Type::BlockParamProxy as usize <= 0b1111);
```

Comment:
Unfortunate we don't have a way[^1](https://github.com/rust-lang/rust/issues/73662) to count the number of variants in a enum to make this robust against adding variants to the end of `Type`, so it may be good to add a comment at the declaration of `Type` to warn against it.

Reply:
Sure. At the moment I did add an assertion in `set_local_type()` to guard against it. Will add a comment as well.